### DISCUSSION TITLE
wazuh-agent: ERROR: Duplicate agent name: computer (from manager)

### DISCUSSION THREAD

**Discussion by marc-ballesteros-tmt** [wazuh/wazuh#26268] [Q&A] [answered]
Hi. We have deployed a wazuh instance on Kubernetes and are testing. I have a problem when adding an agent. I get the following error:

2024/10/10 13:53:10 wazuh-agent: ERROR: Duplicate agent name: computer (from manager)
2024/10/10 13:53:10 wazuh-agent: ERROR: Unable to add agent (from manager)

I have tried several things: Remove the agent with the command /var/ossec/bin/manage_agents from manager. It didn't work.

I have also tried to configure another name in the ossec.conf file of the agent, in the enrollment section:

<agent_name>computer2</agent_name>

And it doesn't work either. Can someone enlighten me? I can't find any solution...
Thank you!

**Comment by Hasitha9796** [comment, marked as answer]
Hi [marc-ballesteros-tmt](https://github.com/marc-ballesteros-tmt)

The error that you have shared is related to the agent duplicate name issue. This typically occurs when a Wazuh agent tries to re-enroll using a different key while retaining the same agent name.

Ensure there is a stable and reliable network connection between the Wazuh Manager and the agents. If the connection is unstable, agents might try to re-enroll as new agents, which can lead to agent name duplication issues.

You can solve this error in different ways:
Delete the existing agent: taking into account the log, there is already a registered agent with the same name, but if you no longer need it or you want to re-register it, you should delete it from the manager in the first place. You can refer to this [link](https://documentation.wazuh.com/current/user-manual/agent/agent-management/remove-agents/restful-api-remove.html) to do so, or execute this command in your manager: `/var/ossec/bin/manage_agents -r <AGENT_ID>`

Set a different name in the enrollment configuration: you can change the name of the agent you want you re-register using its configuration. In this [link](https://documentation.wazuh.com/current/user-manual/agent/agent-enrollment/enrollment-methods/via-agent-configuration/index.html), you have all the information.
Check the document for agent Deployment variables and use the `WAZUH_AGENT_NAME` with a different agent name value while installing the agent.
https://documentation.wazuh.com/current/user-manual/agent/agent-enrollment/deployment-variables/index.html

Using the force option will force the insertion of an agent if there is a duplicate name or IP address and will remove the old agent with the same name or IP address, as explained in our documentation. This will create another problem as your agents will be removed and registered again in a looping process.
https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html#force

If the issue still persists and you need further assistance, please share the following files for analysis:
The Wazuh agent `/var/ossec/etc/ossec.conf `file
The shared `/var/ossec/etc/shared/agent.conf` file (if applicable)
The Wazuh agent `/var/ossec/logs/ossec.log` file from the agent that is showing this issue

Looking forward to your update on the issue.
